gax/src/main/java/com/google/api/gax/retrying/BasicRetryingFuture.java

```diff
@@ -137,14 +138,23 @@ void clearAttemptServiceData() {
 
   // "super." is used here to avoid infinite loops of callback chains
   void handleAttempt(Throwable throwable, ResponseT response) {
+    ApiTracer tracer = retryingContext.getTracer();
+
     synchronized (lock) {
       try {
         clearAttemptServiceData();
         if (throwable instanceof CancellationException) {
           // An attempt triggered cancellation.
+          // In almost all cases, the operation caller caused the attempt to trigger the
+          // cancellation by invoking cancel() on the CallbackChainRetryingFuture, which cancelled
+          // the current attempt.
+          // In a theoretical scenario, the attempt callable might've thrown the exception on its
+          // own volition. However it's currently impossible to disambiguate the 2 scenarios.
+          tracer.attemptCancelled();
```

Comment:
Won't storing cancelation message be beneficial here?

Reply:
I kept going back & forth on including the cancellation message. I'll merge this with out it for now. Once I get a bit of experience instrumenting the bigtable client, I'll come back to this